controllers/configurationpolicy_controller.go

```diff
@@ -921,7 +950,10 @@ func addConditionToStatus(
 
 	var complianceState policyv1.ComplianceState
 
-	if compliant {
+	if reason == "Error cleaning up child objects" {
```

Comment:
Nit: I think this string could be better as a `const` somewhere, as opposed to a magic string.